### DISCUSSION TITLE
Page title not being translated after deployment (on Vercel)

### DISCUSSION THREAD

**Discussion by CHR-onicles** [sergiodxa/remix-i18next#219] [Q&A]
Hi,

I've followed the instructions from the readme on using `getFixedT()` in my loader and returning translated text which I then use in my meta function. This works in my local dev and preview environments, but I've noticed on Vercel it fails to translate, and just displays the translation keys:

Local env:
![image](https://github.com/user-attachments/assets/cc043cd2-d99f-441f-a74c-92a41003d335)

Deployed version:
![image](https://github.com/user-attachments/assets/79e6d9d2-c99c-4564-976b-e5d2583bf730)

I've tried all sort of different configs but it doesn't seem to work. In Vercel logs, I see the accepted languages from logging `request` but I'm starting to think i18n might not be ready early enough to translate stuff in the `<head>` element? Everything else on the page works absolutely fine!

I'll appreciate any help with this 🙏🏼

**Comment by sergiodxa** [comment]
The most probable issue is that the translation files are not being loaded server side, this is common in Vercel if you use the fs backend of i18next, I recommend you to import them server side or fetch them with another backend

**Reply by CHR-onicles** [reply]
Hi @sergiodxa, Thanks for the response!

I've tried removing the backend and importing translations directly through resources. However, now even locally the page title is still not translated but the rest of the page is translated without issues. This is my current setup to highlight the changes. Again, using the readme as the base template:

```ts
// i18n.server.ts
import { RemixI18Next } from "remix-i18next/server";
import i18n from "~/localization/i18n"; // Main config file

const i18next = new RemixI18Next({
  detection: {
    supportedLanguages: i18n.supportedLngs,
    fallbackLanguage: i18n.fallbackLng,
    async findLocale(request) {
      const { pathname } = new URL(request.url);
      const locale = pathname.split("/").at(1) || null;
      return locale;
    },
  },
  i18next: {
    ...i18n,
  },
});

export default i18next;

```


```ts
// entry.client.tsx
import i18n from "./localization/i18n";
import en from "./localization/en/translation.json";
import fr from "./localization/fr/translation.json";

async function hydrate() {
  await i18next
    .use(initReactI18next)
    .use(LanguageDetector)
    .init({
      ...i18n,
      ns: getInitialNamespaces(),
      resources: { en: { translation: en }, fr: { translation: fr } },
      detection: {
        order: ["htmlTag"],
        caches: [],
      },
    });
    
    // ...rest of code here same as in readme
    }
```


```ts
// entry.server.ts
import i18next from "~/localization/i18n.server";
import i18n from "./localization/i18n";
import en from "./localization/en/translation.json";
import fr from "./localization/fr/translation.json";

const ABORT_DELAY = 5000;

export default async function handleRequest(
  request: Request,
  responseStatusCode: number,
  responseHeaders: Headers,
  remixContext: EntryContext
) {
  const callbackName = isbot(request.headers.get("user-agent"))
    ? "onAllReady"
    : "onShellReady";

  const instance = createInstance();
  const lng = await i18next.getLocale(request);
  const ns = i18next.getRouteNamespaces(remixContext);

  await instance.use(initReactI18next).init({
    ...i18n,
    lng,
    ns,
    resources: { en: { translation: en }, fr: { translation: fr } },
  });
  
  //...rest of code here same as in readme
```

```ts
// page loader
import i18nextServer from "~/localization/i18n.server";


export const loader = async ({ request }: LoaderFunctionArgs) => {
  const t = await i18nextServer.getFixedT(request);

  const title = `${t("book-appointment-now")} • ${PAGE_TITLE}`;

  return json({ title });
};
```

```ts
// page meta function
export const meta: MetaFunction<typeof loader> = ({ data }) => {
  console.log("META TITLE:", data?.title);
  return [{ title: data?.title }, { name: "og:title", content: data?.title }];
};

```


Result locally and deployed:
![image](https://github.com/user-attachments/assets/995c5b50-cbe1-403b-941c-b25ed7263477)


Is there something I'm missing?

**Reply by CHR-onicles** [reply]
Hey @sergiodxa, I did what you recommended with no luck, any other suggestions?

**Reply by sergiodxa** [reply]
If you can create a reproduction I could try to look into it.

Also double check that your `findLocale` method is returning a valid locale

**Reply by CHR-onicles** [reply]
@sergiodxa, [Link to a minimal reproduction repo](https://github.com/CHR-onicles/remix-i18next-title-bug-reproduction)

I've also deployed it on Vercel.

I import translations server side as you suggested.

**Reply by CHR-onicles** [reply]
Hi @sergiodxa, I noticed you closed this discussion, has this issue been fixed? I just updated all the packages in this [minimal reproduction](https://github.com/CHR-onicles/remix-i18next-title-bug-reproduction) and tested it, but still had the same issue.

You can easily see this with this [deployment link](https://remix-i18next-title-bug-reproduction-47gfu7f6z.vercel.app/), The title just shows the translation key, without being actually translated.

**Reply by sergiodxa** [reply]
I closed it because it was too old, I never use Vercel so I haven' really tried, on CF Workers and long running Node I can't reproduce it
